packages/admin/cms-admin/src/pages/pageTree/PageLabel.tsx

```diff
@@ -14,28 +14,49 @@ interface PageLabelProps {
     onClick?: (e: React.MouseEvent) => void;
 }
 
+function createMatches(page: PageTreePage): {
+    pageMatches: TextMatch[];
+    pathMatches: TextMatch[];
+} {
+    const pageMatches: TextMatch[] = [];
+    const pathMatches: TextMatch[] = [];
+
+    page.matches.forEach((match) => {
+        if (match.where === "name") {
+            pageMatches.push(match);
+        } else if (match.where === "path") {
+            pathMatches.push(match);
+        }
+    });
+
+    return { pageMatches, pathMatches };
+}
+
 const PageLabel: React.FunctionComponent<PageLabelProps> = ({ page, disabled, onClick }) => {
     const { documentTypes } = usePageTreeContext();
     const documentType = documentTypes[page.documentType];
+    const { pageMatches, pathMatches } = createMatches(page);
 
     return (
         <Root onClick={onClick}>
             <PageTypeIcon page={page} disabled={disabled} />
             <LinkContent>
                 <LinkText color={page.visibility === "Unpublished" || disabled ? "textSecondary" : "textPrimary"}>
-                    <MarkedMatches text={page.name} matches={page.matches} />
+                    <MarkedMatches text={page.name} matches={pageMatches} />
```

Comment:
Why not use `<MarkedMatches text={page.name} matches={page.matches.filter(match => match.where === "name"}} />` and `<MarkedMatches text={page.name} matches={page.matches.filter(match => match.where === "path"}} />` instead of creating two additional arrays and an additional function?